external_parser/event_processors/joined_event.h

```diff
@@ -18,7 +18,8 @@ struct typed_joined_event {
   virtual ~typed_joined_event() = default;
   virtual bool is_skip_learn() const = 0;
   virtual void set_skip_learn(bool sl) = 0;
-  virtual bool should_calculate_apprentice_reward() const = 0;
+  virtual bool should_calculate_apprentice_reward(std::vector<unsigned> actions,
```

Comment:
it looks like the `interaction.action` and `interaction.action[i]` (for cb and ccb) and the baselin action are member variables of each typed event (i.e. `cb_joined_event` and `ccb_joined_event`) so the arguments don't need to be passed in here since they can be accessed directly from each `should_calculate_apprentice_reward` function